src/vtok_agent/src/util.rs

```diff
@@ -4,6 +4,32 @@ use std::fmt::Write;
 use std::fs::OpenOptions;
 use std::io::Read;
 use std::path::Path;
+use std::sync::atomic::Ordering;
+use std::time::{Duration, Instant};
+
+use crate::gdata;
+use nix::libc;
+
+pub enum SleepError {
+    UserExit,
+}
+
+pub fn interruptible_sleep(dur: Duration) -> Result<(), SleepError> {
+    let wake_time = Instant::now() + dur;
```

Comment:
nit: 
If the duration is 0 just return.
Also, this function could receive negative parameters.

Reply:
The `Duration` can't be negative, as it's using only unsigned types internally. If it's `0`, the second `match` arm will hit and head straight to return.

However, `Instant::now() + dur` could indeed overflow `u64` if an ungodly `Duration` is passed as an argument. But that would imply the caller is doing something terribly wrong, in which case I think it's ok to panic and force the caller to fix their code.